node/src/accountant/payment_adjuster/preparatory_analyser/mod.rs

```diff
@@ -0,0 +1,665 @@
+        error_factory: ErrorFactory,
+        expected_error_preparer: F,
+    ) where
+        F: FnOnce(usize, u128, u128) -> Error,
+        ErrorFactory: ServiceFeeSingleTXErrorFactory<Error>,
+        Error: Debug + PartialEq,
+    {
+        let mut account_1 = make_analyzed_account(111);
+        account_1.qualified_as.bare_account.balance_wei = 2_000_000_000;
+        account_1.disqualification_limit_minor = 1_500_000_000;
+        let mut account_2 = make_analyzed_account(222);
+        account_2.qualified_as.bare_account.balance_wei = 1_000_050_000;
+        account_2.disqualification_limit_minor = 1_000_000_101;
+        let mut account_3 = make_analyzed_account(333);
+        account_3.qualified_as.bare_account.balance_wei = 1_000_111_111;
+        account_3.disqualification_limit_minor = 1_000_000_222;
+        let cw_service_fee_balance = 1_000_000_100;
+        let supplied_accounts = vec![account_1, account_2, account_3];
+        let supplied_accounts_count = supplied_accounts.len();
+        let service_fee_total_of_the_known_set = 2_000_000_000 + 1_000_050_000 + 1_000_111_111;
+
+        let result = PreparatoryAnalyzer::check_adjustment_possibility(
+            &supplied_accounts,
+            cw_service_fee_balance,
+            error_factory,
+        );
+
+        let expected_error = expected_error_preparer(
+            supplied_accounts_count,
+            service_fee_total_of_the_known_set,
+            cw_service_fee_balance,
+        );
+        assert_eq!(result, Err(expected_error))
+    }
+
+    #[test]
+    fn not_enough_for_even_the_least_demanding_account_error_right_after_positive_tx_fee_check() {
+        let error_factory = EarlyServiceFeeSingleTXErrorFactory::default();
+
+        let expected_error_preparer =
+            |_, total_amount_demanded_in_accounts_in_place, cw_service_fee_balance_minor| {
+                ServiceFeeImmoderateInsufficiency {
+                    total_service_fee_required_minor: total_amount_demanded_in_accounts_in_place,
+                    cw_service_fee_balance_minor,
+                }
+            };
+
+        test_not_enough_for_even_the_least_demanding_account_causes_error(
+            error_factory,
+            expected_error_preparer,
+        )
+    }
+
+    #[test]
+    fn not_enough_for_even_the_least_demanding_account_error_right_after_tx_fee_accounts_dump() {
+        let original_accounts = vec![
+            make_weighed_account(123),
+            make_weighed_account(456),
+            make_weighed_account(789),
+            make_weighed_account(1011),
+        ];
+        let original_number_of_accounts = original_accounts.len();
+        let initial_sum = sum_as(&original_accounts, |account| account.balance_minor());
+        let error_factory = LateServiceFeeSingleTxErrorFactory::new(&original_accounts);
+        let expected_error_preparer = |number_of_accounts, _, cw_service_fee_balance_minor| {
```

Comment:
Let's rename the act here to say,`prepare_expected_error`.

Please fix it in the previous test too.

 - @bertllll